bitbots_vision_tools/scripts/colorpicker.py

```diff
@@ -0,0 +1,254 @@
+                                        ]
+                # Serialize the colors
+                selected_matrix = selected_matrix.reshape(-1,3)
+                # Get the unique colors
+                selected_matrix = np.unique(selected_matrix, axis=0)
+
+                # Copy the latest color space
+                color_space = self._history[-1].copy()
+
+                # Check for left click or shift + left click
+                if self._undo_click:
+                    # Remove color values from color space
+                    color_space[
+                        selected_matrix[:,0],
+                        selected_matrix[:,1],
+                        selected_matrix[:,2]] = 0
+                # Check if we add stuff
+                elif self._left_click:
+                    # Add color values to color space
+                    color_space[
+                        selected_matrix[:,0],
+                        selected_matrix[:,1],
+                        selected_matrix[:,2]] = 1
+                else:
+                    print("This should never happen")
+
+                # Append new color space to self._history
+                self._history.append(color_space)
+
+                # Reset events
+                self._left_click, self._undo_click = False, False
+
+            # Mask the image with the current color space
+            mask = VisionExtensions.maskImg(self._image, self._history[-1])
+
+            # Draw the mask on the canvas
+            canvas = self._draw_mask(canvas, mask, (255,0,255), opacity=0.8)
+
+            # Draw selection area
+            cv2.rectangle(canvas,
+                (box_min_x, box_min_y),
+                (box_max_x, box_max_y),
+                (0, 255, 0),
+                2)
+
+            # Show canvas
+            cv2.imshow("Colorpicker", canvas)
+
+            # Key checks for UI events
+            key = cv2.waitKey(1) & 0xFF
+
+            # Increase selection box size
+            if key == ord("+"):
+                self._box_size += int(self._box_size * 0.2) + 1
+            # Reduce selection box size
+            if key == ord("-") and self._box_size > 1:
+                self._box_size -= int(self._box_size * 0.2) - 1
+            # Undo
+            if key == ord("u") and len(self._history) > 1:
+                del self._history[-1]
+            # Quit exit
+            elif key%256 == 27 or 0xFF == ord('q'):
+                save = False
+                break
+            # Save exit
+            elif key == ord("s") or key == ord("w"):
+                save = True
+                break
+
+            time.sleep(0.01)
+
+        # Close all open windows
+        cv2.destroyAllWindows()
+        return save
+
+    def _save(self):
+        """
+        Saves the current colorspace in a pickle file.
+        """
+        # Get user input
+        input_path = input( "Press enter to save look up table to '{}' "
```

Comment:
On every other occasion, its called color space, so...
```suggestion
        input_path = input( "Press enter to save color space file to '{}' "
```

Reply:
But technically it's a look up table. But we use color space all the time, so it doesn't matter. We can change it.